crates/flox-rust-sdk/src/models/environment.rs

```diff
@@ -33,75 +35,112 @@ pub enum InstalledPackage {
     StorePath(StorePath),
 }
 
-/// Struct representing a local environment in a given location and state
+#[async_trait(?Send)]
+pub trait Environment {
+    type ConcreteTemporary;
```

Comment:
```suggestion
    type TempEnv;
```

This will already be a concrete type, no need to call it concrete.

Reply:
We already have a `TemporaryEnvironment` - open to other names though